package/kernel/mwlwifi/patches/004-Fix-debugfs-compilation-warning-in-mwl_debugfs_info_read.patch

```diff
@@ -0,0 +1,28 @@
+From 44419b1feae3eedda21cdc71da9acb611ca1a6fd Mon Sep 17 00:00:00 2001
+From: Christian Marangi <ansuelsmth@gmail.com>
+Date: Sun, 12 May 2024 14:42:01 +0200
+Subject: [PATCH] Fix debugfs compilation warning in mwl_debugfs_info_read
+
+Fix debugfs compilation warning on 64Bit mwl_debugfs_info_read by
+casing for uintptr_t and use %zx.
+
+Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
+---
+ debugfs.c | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+--- a/debugfs.c
++++ b/debugfs.c
+@@ -364,10 +364,10 @@ static ssize_t mwl_debugfs_info_read(str
+ 			 "-----------------------=>  address| address|qlen|fw_desc_cnt\n");
+ 		spin_lock_irqsave(&pcie_priv->tx_desc_lock, flags);
+ 		len += scnprintf(p + len, size - len,
+-				"wcb_base0   : %x => %8x|%8p|%4d|%d\n", get_hw_spec->wcb_base0, *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),skb_queue_len(&pcie_priv->txq[0]),pcie_priv->fw_desc_cnt[0]);
++				"wcb_base0   : %x => %8zx|%8p|%4d|%d\n", get_hw_spec->wcb_base0, *((size_t *)(size_t)le32_to_cpu(get_hw_spec->wcb_base0)),(void *)*((uintptr_t *)(uintptr_t)le32_to_cpu(get_hw_spec->wcb_base0)),skb_queue_len(&pcie_priv->txq[0]),pcie_priv->fw_desc_cnt[0]);
```

Comment:
@Ansuel 
Dereferencing a 32 bit value as a 64 pointer on a 64 bit CPU will crash. It would be better to conditionally compile the two lines and exclude them on 64 bit CPUs.

@robimarko @neheb

Reply:
Perhaps the following could be a solution
`
if (priv->chip_type == MWL8864) {
    len += scnprintf(p + len, size - len,
        "-----------------------=>  address| address|qlen|fw_desc_cnt\n");

    spin_lock_irqsave(&pcie_priv->tx_desc_lock, flags);

    // Common base address for wcb_base0
    uint32_t wcb_base0_cpu = le32_to_cpu(get_hw_spec->wcb_base0);
    const char *pointer_msg = " (no pointer on 64-bit) |";
    
    // Handle wcb_base0 for 32-bit and 64-bit CPUs
#ifdef CONFIG_64BIT
    len += scnprintf(p + len, size - len,
        "wcb_base0   : %x => %08x|%s%4d|%d\n", wcb_base0_cpu, wcb_base0_cpu, pointer_msg, skb_queue_len(&pcie_priv->txq[0]), pcie_priv->fw_desc_cnt[0]);
#else
    len += scnprintf(p + len, size - len,
        "wcb_base0   : %x => %08x|%p|%4d|%d\n", wcb_base0_cpu, wcb_base0_cpu, (void *)(uintptr_t)wcb_base0_cpu, skb_queue_len(&pcie_priv->txq[0]), pcie_priv->fw_desc_cnt[0]);
#endif

    // Loop over the remaining wcb_base[i] for all queues
    for (i = 0; i < SYSADPT_TOTAL_TX_QUEUES - 1; i++) {
        uint32_t wcb_base_cpu = le32_to_cpu(get_hw_spec->wcb_base[i]);
        
#ifdef CONFIG_64BIT
        len += scnprintf(p + len, size - len,
            "wcb_base[%2d]: %x => %08x|%s%4d|%d\n", i, wcb_base_cpu, wcb_base_cpu, pointer_msg, skb_queue_len(&pcie_priv->txq[i + 1]), pcie_priv->fw_desc_cnt[i + 1]);
#else
        len += scnprintf(p + len, size - len,
            "wcb_base[%2d]: %x => %08x|%p|%4d|%d\n", i, wcb_base_cpu, wcb_base_cpu, (void *)(uintptr_t)wcb_base_cpu, skb_queue_len(&pcie_priv->txq[i + 1]), pcie_priv->fw_desc_cnt[i + 1]);
#endif
    }

    spin_unlock_irqrestore(&pcie_priv->tx_desc_lock, flags);
}
`